pkg/interactor/list_data_test.go

```diff
@@ -0,0 +1,140 @@
+package interactor
+
+import (
+	"errors"
+	"testing"
+
+	thingentities "github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+type ListDataTestCase struct {
+	name               string
+	authorization      string
+	query              *entities.Query
+	expectedData       []entities.Data
+	expectedDataError  error
+	expectedThing      []*thingentities.Thing
+	expectedThingError error
+	fakeLogger         *mocks.FakeLogger
+	fakeDataStore      *mocks.FakeDataStore
+	fakeService        *mocks.FakeService
+}
+
+var ldCases = []ListDataTestCase{
```

Comment:
The `list data` is a complex operation that can return data according to three scenarios:

- list all data (no `thingID` or `sensorID` are passed)
- list data by thing (no `sensorID` is passed)
- list data by sensor

You should test all of them.

Reply:
Done!